src/lightning_app/storage/filesystem.py

```diff
@@ -0,0 +1,113 @@
+import os
+import shutil
+from pathlib import Path
+from typing import Callable
+
+from fsspec.implementations.local import LocalFileSystem
+
+from lightning_app.storage.copier import _copy_files
+from lightning_app.storage.path import _filesystem, _shared_storage_path
+
+
+def _get_files(self, fs, src: Path, dst: Path, overwrite: bool):
+    if fs.isdir(src):
+        if isinstance(fs, LocalFileSystem):
+            dst = dst.resolve()
+            if fs.exists(dst):
+                if overwrite:
+                    fs.rm(str(dst), recursive=True)
+                else:
+                    raise FileExistsError(f"The file {dst} was found. Add get(..., overwrite=True) to replace it.")
+
+            shutil.copytree(src, dst)
+        else:
+            glob = f"{str(src)}/**"
+            fs.get(glob, str(dst.absolute()), recursive=False)
+    else:
+        fs.get(str(src), str(dst.absolute()), recursive=False)
+
+
+class FileSystem:
+
+    """This filesystem enables to easily move files from and to the shared storage."""
+
+    def __init__(self):
+        self._fs = _filesystem()
+        self._root = str(_shared_storage_path())
+
+    def _validate_path(self, src_path: str, dst_path: str) -> None:
+        if not os.path.exists(Path(src_path).resolve()):
+            raise FileExistsError(f"The provided path {src_path} doesn't exists")
+
+        if str(Path(dst_path).resolve()) == dst_path:
+            raise RuntimeError(f"The destination path {dst_path} should be relative.")
+
+    def put(self, src_path: str, dst_path: str, put_fn: Callable = _copy_files) -> None:
+        """This method enables to put a file to the shared storage in a blocking fashion.
+
+        Arguments:
+            src_path: The path to your files locally
+            dst_path: The path to your files transfered in the shared storage.
+            put_fn: The method to use to put files in the shared storage.
+        """
+        self._validate_path(src_path, dst_path)
```

Comment:
Is `dst_path` forced to be relative? In my opinion we should require `dst_path` to be absolute with the `lit://` prefix, personally I find relative confusing (relative to what?)

Reply:
Of course it would have the impression of being absolute, since in the backend it's always relative to where storage is for the app, but that's a detail.

Reply:
Hey @lantiga. I removed the limitations. Now, it is free for the users.